Common/src/linear_algebra/CSysMatrixGPU.cu

```diff
@@ -26,34 +35,251 @@
  */
 
 #include "../../include/linear_algebra/CSysMatrix.hpp"
+#include "../../include/geometry/CGeometry.hpp"
 #include "../../include/linear_algebra/GPUComms.cuh"
 
+using namespace cudaKernelParameters;
+
+template<typename matrixType, typename vectorType>
+__device__ void DeviceGaussElimination(matrixType* matrixCopy, vectorType* prod, unsigned long row, unsigned int threadNo, bool rowInPartition, matrixParameters matrixParam)
+{
+   auto matrixIndex = [=](unsigned short x, unsigned short y){
+      return (x * matrixParam.blockColSize + y);
+   };
+
+   auto vectorIndex = [=](unsigned short row, unsigned short elemNo){
+      return (row * matrixParam.blockRowSize + elemNo);
+   };
+
+   unsigned short x = threadNo/matrixParam.blockRowSize;
+   unsigned short y = threadNo % matrixParam.blockColSize;
```

Comment:
add some comments to what these are (block row and block column?), x and y are somewhat unusual names for index variables